### PR TITLE
Added link to extension naming conventions

### PR DESCRIPTION
**Pull request by lukasdoerr**

### CODE REVIEW

**Review by brotkrueml — DISMISSED** [review]
Thank you for your contribution.

Personally, I would link to this page for the requirements:
https://docs.typo3.org/m/typo3/reference-coreapi/11.5/en-us/ExtensionArchitecture/BestPractises/ExtensionKey.html#rules-for-the-extension-key
These are the hard requirements. The "naming conventions" are - well - conventions, you can stick to them but you don't have to. What do you think?

Additionally, please don't use an external link. Instead use a reference:
```
:ref:`extension key requirements <extension-key>`
```
(for linking to the rules). This way, the same version of the manual is linked and the real link gets updated when the page is moved around. You can find the reference with "view source" in the generated docs. It starts with `.. _`.

**Comment by brotkrueml** [comment]
For references see also here: https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/WritingReST/CheatSheet.html#headlines

File: Documentation/ExtensionArchitecture/HowTo/PublishExtension/PublishToTER/Index.rst [hunk 1]
```diff
@@ -14,8 +14,8 @@ First of all ask yourself some questions before publishing or even putting some
 effort in coding:
 
 *  What additional benefit does your extension have for the TYPO3 community?
-*  Does your extension key describe the extension? See the extension key
-   requirements.
+*  Does your extension key describe the extension? See the `extension key
+   requirements<https://docs.typo3.org/m/typo3/reference-coreapi/11.5/en-us/ExtensionArchitecture/BestPractises/NamingConventions.html>`__.
```
Comment (linawolf) [inline Documentation/ExtensionArchitecture/HowTo/PublishExtension/PublishToTER/Index.rst:18]:
```suggestion
*  Does your extension key describe the extension? See the :ref:`extension key
   requirements <extension-key>`
```

**Review by linawolf — DISMISSED** [review]

File: Documentation/ExtensionArchitecture/HowTo/PublishExtension/PublishToTER/Index.rst [hunk 2]
```diff
@@ -14,8 +14,8 @@ First of all ask yourself some questions before publishing or even putting some
 effort in coding:
 
 *  What additional benefit does your extension have for the TYPO3 community?
-*  Does your extension key describe the extension? See the extension key
-   requirements.
+*  Does your extension key describe the extension? See the :ref:`extension key
+   requirements <extension-key>`
```
Comment (linawolf) [inline Documentation/ExtensionArchitecture/HowTo/PublishExtension/PublishToTER/Index.rst:18]:
```suggestion
   requirements <extension-key>`.
```

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**